mono/metadata/metadata.c

```diff
@@ -5276,6 +5291,21 @@ do_mono_metadata_type_equal (MonoType *t1, MonoType *t2, gboolean signature_only
 	if (t1->type != t2->type || t1->byref != t2->byref)
 		return FALSE;
 
+	if (t1->num_mods != t2->num_mods)
+		return FALSE;
+
+	for (int i=0; i < t1->num_mods; i++) {
+		MonoError error;
+		MonoClass *c1 = mono_class_get_checked (t1->modifiers [i].image, t1->modifiers->exported.token, &error);
+		mono_error_assert_ok (&error);
+		MonoClass *c2 = mono_class_get_checked (t2->modifiers [i].image, t2->modifiers->exported.token, &error);
+		mono_error_assert_ok (&error);
+
+		if (c1 != c2)
```

Comment:
What if the two signatures have the modifiers in different orders? (Is there an ordering rule in ECMA?)

Reply:
That's worrying, let me look into that.